### ISSUE TITLE
ghc build failure with recent sphinx-doc

### ISSUE THREAD

**Issue by timotheecour** [Homebrew/homebrew-core#33249]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
only warnings

- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
https://gist.github.com/timotheecour/7812f64f07b0147b1c39a5e045b8f150

- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

Not sure whether related to:
* https://github.com/Homebrew/homebrew-core/issues/25568 
* [ghc 8.6.1 by commitay · Pull Request #32342 · Homebrew/homebrew-core](https://github.com/Homebrew/homebrew-core/pull/32342)

**Comment by Rotonen** [comment]
Relevant bit (from my machine, as this fails for me locally too):

```
/private/tmp/ghc-20181022-90728-f22e84/ghc-8.4.3/utils/haddock/doc/invoking.rst:457: WARNING: unknown option: -cpp
generating indices... genindex
writing additional pages... search
copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 1 warning.

The HTML pages are in .build-html.
cp -R utils/haddock/doc/.build-html utils/haddock/doc/haddock
make: *** [all] Error 2
```

Despite that being with relative paths, could having homebrew in `$HOME` be a problem for how the docs get built and installed by this formula?

**Comment by fxcoudert** [comment]
Apparently a bug with recent `sphinx-doc` versions: https://phabricator.haskell.org/D5161
A fix is available, too…

**Comment by apjanke** [comment]
Looks like https://github.com/Homebrew/homebrew-core/pull/33482 will fix.

**Comment by fxcoudert** [comment]
#33482 is a fix for ghc 8.4.3. It was merged in https://github.com/Homebrew/homebrew-core/commit/fa66e1a5b375970d181fe3555272754d37941742
